Source/WebCore/rendering/style/RenderStyle.cpp

```diff
@@ -1356,6 +1356,23 @@ bool RenderStyle::changeRequiresRecompositeLayer(const RenderStyle& other, Optio
     return false;
 }
 
+inline static bool mayRequireOverflowScrolling(const RenderStyle& style)
+{
+    return style.overflowX() == Overflow::Scroll || style.overflowX() == Overflow::Auto
+        || style.overflowY() == Overflow::Scroll || style.overflowY() == Overflow::Auto;
+}
+
+bool RenderStyle::changeRequiresSimplifiedLayout(const RenderStyle& other, OptionSet<StyleDifferenceContextSensitiveProperty>&) const
+{
+    if (mayRequireOverflowScrolling(*this) && mayRequireOverflowScrolling(other)) {
```

Comment:
I wish there was a diff value in-between StyleDifference::SimplifiedLayout and StyleDifference::RepaintLayer, something along the lines of "skip layout but run a full layer update" and initiate that unconditionally on visibility change (I am somewhat convinced this helps with resetting some stale cached states in a non-scrolling case too)

Reply:
Oh, interesting — I had thought that "skip layout but run a full layer update" was already basically `SimplifiedLayout`, but I suppose `SimplifiedLayout` still triggers some kind of layout... Are you proposing something like a `RepaintLayerAndRecomputeOverflow`?

Reply:
(Filed https://bugs.webkit.org/show_bug.cgi?id=258522 for this)